packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/README.md

```diff
@@ -0,0 +1,377 @@
+# Technical documentation
+
+Here we will cover how the `<Permissions />` component works.
+This component is in charge of managing the permissions of the admin panel, it has 4 children one child for each tab.
+The collection types & single types tabs uses the same component: `<ContentTypes>` similarly, the settings & plugins tab uses the `<Plugins>` component.
+
+The UI uses the layout received from the back-end in order to build the UI.
+
+## Layout shape
+
+```js
+const layout = {
+  conditions: [{ id: 'string', displayName: 'string', category: 'string'}], // Array of conditions that can be applied on a permission
+  sections: {
+    plugins: [
+      { displayName: 'string', action: 'string', subCategory: 'string', plugin: 'string'}
+    ],
+    settings: [], // Same shape as the plugins
+    collectionTypes: {
+      subjects: [ // Array of subjects
+        {
+          uid: 'string', // Collection type uid,
+          label: 'string', // Collection type label,
+          properties: [
+            {
+              label: 'string', // Property label ex: Fields,
+              value: 'string', // Value of the property ex: fields,
+              children: [ // This corresponds to the fields that we will display
+                {
+                  label: 'string',
+                  value: 'string',
+                  required: 'boolean', // This key is optional,
+                  children: [], // This key is optional, if it exists it means that a field has nested fields
```

Comment:
Do you think that there might be confusion between this children and the one we have in programmatic APIs?

Reply:
Yes! this is why in the utilities I have renamed this key to `childrenForm`